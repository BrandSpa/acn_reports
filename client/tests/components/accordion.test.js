import React from "react";
import { shallow } from "enzyme";
import Accordion from "../../components/accordion";

describe('accordion component', () => {
	it('should render as expected', () => {
		let wrapper = shallow(<Accordion content={'this is his content'} btnTitle='acn int' />);
		expect(wrapper.getNodes()).toMatchSnapshot();
	})

	it('it should toggle to true', () => {
		let wrapper = shallow(<Accordion />);
		wrapper.find('button').simulate('click');
		expect(wrapper.state().show).toBe(true);
	})

	it('it should have icon up', () => {
			let wrapper = shallow(<Accordion />);
		wrapper.find('button').simulate('click');
		expect(wrapper.find('button i').hasClass('ion-chevron-up')).toBe(true);
	})

	it('it should toggle to false', () => {
		let wrapper = shallow(<Accordion />);
		wrapper.find('button').simulate('click');
		wrapper.find('button').simulate('click');
		expect(wrapper.state().show).toBe(false);
	})

	it('it should have icon down', () => {
			let wrapper = shallow(<Accordion />);
		wrapper.find('button').simulate('click');
		wrapper.find('button').simulate('click');
		expect(wrapper.find('button i').hasClass('ion-chevron-down')).toBe(true);
	})
})
