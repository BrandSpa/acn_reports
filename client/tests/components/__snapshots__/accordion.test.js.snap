// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accordion component should render as expected 1`] = `
Array [
  <div
    className="accordion"
    data-jsx={800370740}
>
    <button
        className="accordion__btn"
        data-jsx={800370740}
        onClick={[Function]}
        style={
            Object {
                "background": "#687f87",
                "color": "",
              }
        }
    >
        acn int
         
        <i
            className="ion-chevron-down"
            data-jsx={800370740}
        />
    </button>
    <div
        className="accordion__content"
        data-jsx={800370740}
        style={
            Object {
                "display": "none",
              }
        }
    >
        <div
            dangerouslySetInnerHTML={
                Object {
                    "__html": "this is his content",
                  }
            }
            data-jsx={800370740}
        />
    </div>
    <_class
        css=".accordion__btn[data-jsx=\\"800370740\\"]{width:100%, height:60px, border:none, border-radius:0, font-size:18px, font-weight:normal,}"
        styleId={800370740}
    />
</div>,
]
`;
